### PR TITLE
pin-1178: Risk analysis template

### PR DESCRIPTION
**Pull request by galales**

### CODE REVIEW

File: src/main/resources/riskAnalysisTemplate/index.html [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        <img src="./assets/media/logo_pagopa.png" />
+      </div>
+    </div>
+
+
+    <!-- Titolo del documento -->
+    <h1>Analisi del rischio</h1>
+
+        <!-- Intro legale -->
+        <div class="legal-intro">Ai sensi dell’art. 26, comma 11, del decreto-legge, PagoPA s.p.a. nella sua qualità di gestore ex lege della Piattaforma Interoperabilità PDND di cui allo stesso art. 26, con ogni valore legale per l'opponibilità a terzi, ATTESTA CHE [TBC Lorem ipsum]:</div>
```
Comment (galales) [inline src/main/resources/riskAnalysisTemplate/index.html]: Still TBD

File: src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeVersionActivation.scala [hunk 2]
```diff
@@ -162,12 +164,22 @@ final case class PurposeVersionActivation(
...
+    purpose: Purpose,
+    version: PurposeVersion,
+    stateChangeDetails: StateChangeDetails,
+    eService: EService
+  )(implicit contexts: Seq[(String, String)]): Future[PurposeVersion] = {
     val documentId: UUID = uuidSupplier.get
     for {
-      path <- createRiskAnalysisDocument(documentId, purpose, version)
+      producer <- partyManagementService.getInstitutionById(eService.producerId)
+      consumer <- partyManagementService.getInstitutionById(purpose.consumerId)
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeVersionActivation.scala]: I'm leaving here a suggestion that is not strictly a review comment to implement. To the best of my understanding, these 2 calls are orthogonal and could be executed in parallel, right? Why don't we perform it like 
```scala
(producer,consumer) <- Future.sequence(
  partyManagementService.getInstitutionById(eService.producerId),
  partyManagementService.getInstitutionById(purpose.consumerId)
)
```
IMHO we should consider adding a few combinators for Future in commons

Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeVersionActivation.scala]: In this particular case, this is totally yak shaving, i do agree @galales :D

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeVersionActivation.scala (hunk 2 above)
Comment (galales) [reply src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeVersionActivation.scala]: optimizations are always welcome.
Just a note: `Future.sequence` takes an iterable, so the code would be
```
      List(producer, consumer) <- Future.sequence(
        List(
          partyManagementService.getInstitutionById(eService.producerId),
          partyManagementService.getInstitutionById(purpose.consumerId)
        )
      )
```
without compile check on the elements of the result list (I would avoid pattern matching here).
Not ideal, but performance optimizations rarely are.

If we all agree on this solution I'll add it.

Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeVersionActivation.scala]: You're right indeed. I remember that `zip` and `zipWith` combinators should exist on Future. I'll check locally whether they're sequential or parallel and I'll let you know.

Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeVersionActivation.scala]: Local tests and the fact that in these signatures `that` is strict

```scala
def zip[U](that: Future[U]): Future[(T, U)]
def zipWith[U, R](that: Future[U])(f: (T, U) => R)(implicit executor: ExecutionContext): Future[R]
```
makes me think that these futures' execution happens in parallel. I encourage you to test further, though.

Comment (galales) [reply src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeVersionActivation.scala]: confirm that `zip` is parallel 👍

**Review by TonioGela — APPROVED** [review]
🥳

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**